enterprise/cmd/executor/internal/apiclient/queue/client.go

```diff
@@ -79,16 +79,18 @@ func (c *Client) Dequeue(ctx context.Context, workerHostname string, extraArgume
 	return job, decoded, err
 }
 
-func (c *Client) MarkComplete(ctx context.Context, id int) (_ bool, err error) {
+func (c *Client) MarkComplete(ctx context.Context, job executor.Job) (_ bool, err error) {
```

Comment:
Go API surface suggestion:

```suggestion
func (c *Client) MarkComplete(ctx context.Context, id int, token string) (_ bool, err error) {
```

This makes very explicit which fields are required, whereas passing the full executor.Job does not really convey what parts of that big struct are actually required here.

Reply:
Marking resolve as this falls into the discussion on the changes to `workerutl.Store`